apps/server/src/controllers/userControllers/authUsers.ts

```diff
@@ -291,3 +299,62 @@ export const forgotPassword = async (req: Request, res: Response) => {
     })
   }
 }
+
+export const resetPassword = async (req: Request, res: Response) => {
+  try {
+    const { verify: longString } = req.query
+
+    const { newPassword } = req.body
+
+    if (!newPassword) {
+      return res.status(HTTP_STATUS_CODE.BAD_REQUEST).json({
+        message: 'Invalid or missing new-password in the request.',
+      })
+    }
+    // Check for reset-token instance using query-string(id)
+    const resetTokenInstance = await UserResetPasswordToken.findOne({
+      where: { id: longString as string, used: false },
+    })
+
+    if (!resetTokenInstance) {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).json({
+        message: 'Invalid or already used verification token.',
+      })
+    }
+
+    const { id, token } = resetTokenInstance
+
+    /**
+     * References JWT secret in forgotPassword function
+     * @see forgotPassword
+     **/
+    const tokenStatus = await Jwt.isTokenExpired(token, Env.JWT_SECRET + id)
+
+    if (!tokenStatus.valid || tokenStatus.expired) {
+      return res.status(HTTP_STATUS_CODE.UNAUTHORIZED).json({
+        message: 'Invalid or expired reset password token.',
+      })
+    }
+
+    const decodedToken = tokenStatus.data
+
+    const hashedPassword = await PasswordHarsher.hash(newPassword)
+
+    await Users.update(
+      { password: hashedPassword },
+      { where: { id: decodedToken.id } }
+    )
+
+    // Invalidate the reset token after it's used
+    await resetTokenInstance.update({ used: true })
```

Comment:
I didn't see where this data was used to verify if the token has been  used before using the token to reset the password.
Also I feel its best to just clear the record once the token has been used because we won't be needing it for anything again after it has been used.

Reply:
I think so too... but if it was reused immediately, there's no way to know